### ISSUE TITLE
CloudinaryAsset for a list of remote images

### ISSUE THREAD

**Issue by hazzyeer** [cloudinary-devs/gatsby-transformer-cloudinary#78]
Hello,
When trying to use images that already exist on Cloudinary, It seems that the plugin does not recognize a list of images with the structure defined on the [documentation](https://www.npmjs.com/package/gatsby-transformer-cloudinary#use-images-already-on-cloudinary), here is an example: 
```
{
  "title": "Foo",
  "images": [
    {
      "cloudinaryAssetData": true,
      "cloudName": "bar",
      "publicId": "bar/whatever-1",
      "originalHeight": 969,
      "originalWidth": 546
    },
    {
      "cloudinaryAssetData": true,
      "cloudName": "bar",
      "publicId": "bar/whatever-2",
      "originalHeight": 969,
      "originalWidth": 546
    }
  ]
}
```
But if there is only one image, it works as expected : 
```
{
   ...,
  "image": {
      "cloudinaryAssetData": true,
      "cloudName": "bar",
      "publicId": "bar/whatever-1",
      "originalHeight": 969,
      "originalWidth": 546
  }
}
```
I can't even query for `cloudName`, `publicId`, etc.
A possible solution for this?

**Comment by Chuloo** [comment]
thanks @hazzyeer for sending this in. I'm checking it out.

**Comment by kurttomlinson** [comment]
@hazzyeer I think this was fixed in [version 2.1.0](https://github.com/cloudinary-devs/gatsby-transformer-cloudinary/blob/master/packages/gatsby-transformer-cloudinary/CHANGELOG.md#version-210). Deeply nested asset data wasn't processed before that version.

**Comment by kurttomlinson** [comment]
@hazzyeer Actually, I think the issue is that your image nodes are nested inside an array. This plugin doesn't currently look inside arrays for asset data. I'd recommend you name each of your images:

```
{
  "title": "Foo",
  "whateverOne": {
    "cloudinaryAssetData": true,
    "cloudName": "bar",
    "publicId": "bar/whatever-1",
    "originalHeight": 969,
    "originalWidth": 546
  },
  "whateverTwo": {
    "cloudinaryAssetData": true,
    "cloudName": "bar",
    "publicId": "bar/whatever-2",
    "originalHeight": 969,
    "originalWidth": 546
  }
}
```

If you'd like to submit a PR to make this plugin look inside arrays for asset data, this PR will be a good starting point to figure out what needs to be modified: https://github.com/cloudinary-devs/gatsby-transformer-cloudinary/pull/57

**Comment by hazzyeer** [comment]
That's a bug and not an enhancement...

**Comment by raae** [comment]
This should work with beta that is out now. 
Take a look at #163 for how to help with testing.
It would be much appreciated if you would help test the new beta version, it also adds Gatsby v4 support @hazzyeer and @kurttomlinson.

**Comment by cloudinary-raae** [comment]
Solved in v3.0.0
